static/js/unread_ops.js

```diff
@@ -47,17 +55,84 @@ function process_newly_read_message(message, options) {
+        anchor: message_id,
+        include_anchor: include_message,
+        num_before: 0,
+        num_after: NUM_OF_MESSAGES_UPDATED_PER_BATCH,
+        narrow: JSON.stringify(message_lists.current.data.filter.operators()),
+        op: "remove",
+        flag: "read",
+    };
+    channel.post({
+        url: "/json/messages/flags/narrow",
+        data: opts,
+        success(data) {
+            if (!data.found_newest) {
+                ui_report.loading(
+                    $t_html(
+                        {
+                            defaultMessage:
+                                "Working... {messages_marked_unread_till_now} messages marked as unread so far.",
+                        },
+                        {messages_marked_unread_till_now},
+                    ),
+                    $("#request-progress-status-banner"),
+                );
+                if (!loading_indicator_displayed) {
+                    loading.make_indicator(
+                        $("#request-progress-status-banner .loading-indicator"),
+                        {abs_positioned: true},
+                    );
+                    loading_indicator_displayed = true;
+                }
+                messages_marked_unread_till_now += data.affected_count;
+                mark_as_unread_from_here(
+                    data.last_affected_id,
+                    false,
+                    messages_marked_unread_till_now,
+                );
+            } else {
+                if (messages_marked_unread_till_now > 0) {
+                    // messages_marked_unread_till_now is only positive if our request wasn't completed
+                    // in one try and then we displayed the loading banner. So, we now display the success banner too.
+                    messages_marked_unread_till_now += data.affected_count;
+                    loading_indicator_displayed = false;
+                    ui_report.loading(
+                        $t_html(
+                            {
+                                defaultMessage:
+                                    "Done! {messages_marked_unread_till_now} messages marked as unread.",
+                            },
+                            {messages_marked_unread_till_now},
+                        ),
+                        $("#request-progress-status-banner"),
+                        true,
+                    );
+                }
+            }
+        },
+        error(xhr) {
+            // If we hit the rate limit, just continue without showing any error.
+            if (xhr.responseJSON.code === "RATE_LIMIT_HIT") {
+                setTimeout(
+                    () =>
+                        mark_as_unread_from_here(
+                            message_id,
+                            false,
+                            messages_marked_unread_till_now,
+                        ),
+                    xhr.responseJSON["retry-after"],
+                );
+            }
```

Comment:
I think we need an else clause to show some sort of failure message.

Reply:
Ok, added a blueslip error.